raiden/tasks.py

```diff
@@ -144,6 +144,8 @@ def _run(self):  # pylint: disable=method-hidden,too-many-locals
 
             # someone down the line timedout / couldn't proceed
             elif isinstance(response, (RefundTransfer, TransferTimeout)):
+                if hashlock in channel.partner_state.locked:
+                    channel.partner_state.locked.remove(hashlock)
```

Comment:
This doesn't seem right, needs to be checked.

For the initiator it's okay to just discard a `hashlock`, it wouldn't lose asset because of this because it knows that the secret was never revealed, but any mediator needs to keep the `hashlock` registered just in case it was lied to, that means a mediator won't remove the `haslock` and the initiator and first mediator locksroots will out-of-sync drawing the channel useless.